src/main/java/com/devotedmc/ExilePearl/core/CorePearlManager.java

```diff
@@ -287,22 +287,20 @@ public void decayPearls() {
 			}
 
 			PearlHolder holder = pearl.getHolder();
-			if(holder != null && holder.isBlock() && disallowedWorlds.contains(holder.getLocation().getWorld().getName())) {
-				pearlApi.log("Freeing pearl for player %s because the pearl is stored in disallowed world %s.", pearl.getPlayerName(),holder.getLocation().getWorld().getName());
-				pearlsToFree.add(pearl);
-			}
-
 			if (decayTimeout > 0 && pearl.getPlayer() != null) {
 				// player is online now!
 				pearl.setLastOnline(new Date());
 			}
 
 			// convert timeout to milliseconds and compare against last time online.
 			if (decayTimeout == 0 || (new Date()).getTime() - pearl.getLastOnline().getTime() < (decayTimeout * 60 * 1000)) {
+				pearlApi.log("Applying decay to pearl health");
```

Comment:
This log message and the one below are useless because they do not contain the name of the player and will just be printed 300 times. Keep the "set pearl of health of X to  Y" and remove the first one